### ISSUE TITLE
[Bug] Game crashes on Class change menu.

### ISSUE THREAD

**Issue by LupisLord** [Raytwo/Cobalt#15]
Cobalt version: 0.21.0
Hardware: Switch

Two of my mods are having issues when put together:
Bond Rings as Items Wave 6 (requires Skill xml limit expansion)
Gold Crystal And Banes for all (all 3 versions use the same skill.xml)

Weird Bug: Game crashes on Change Class menu.
To be clear, the game runs, you can fight enemies, you can trigger skills from both mods, but it suddenly crashes when scrolling through the Class Change menu.

I think the issue is specifically caused by merging the two Skill.xml patches. Removing one or the other will prevent the crash.
I removed both and manually created a merged Skill.xml and it works fully functional no crashes. 

Both mods can merge Skill.xml with various other mods without causing the issue. I tested each of the above merging with several other mods simultaneously including:
Smashing
Daggers Reworked
Grind skill replacers
Skill and Class Expansion (v2.5.1)

I don't understand how the Skill.xml file triggers the crash. 
I have created a similar looking issue before, but it was caused by the Job.xml file. I had given one class 4 weapons. I assumed the UI doesn't know how to handle because there is only room for 3 weapon icons.

A handful of seemingly random skills are also broken (shows as blank or missing altogether) by the merge:
Personal skills: Jean, Louis, Lapis, Kagetsu
Jobskill: Thief
BondRing skills: Edelgard, Claude, Lilina

This skills can not be activated, but at least do not crash the game.
What's strange is most of these skills are not modified by either mod, only Ring_Lilina was
It may be that another jobskill was corrupted, but I can't search class change menu. Maybe that jobskill is what's actually causing the crash when it was forced to load at class change. But I don't think its jobskill_thief, because one of my tested mods: Daggers Reworked actually replaced theif's jobskill and it still crashed.

Is there anyway I can get access to the merged xml file Cobalt produced at run time?

**Comment by LupisLord** [comment]
Crash description:
Game freezes for several seconds, then gives error message:
"The software was closed because an error occurred."

Switch screenshot is disabled when error message is visible

**Comment by Raytwo** [comment]
> Is there anyway I can get access to the merged xml file Cobalt produced at run time?

Create a directory called ëngage_patches" at the root of your SD

**Comment by LupisLord** [comment]
Comparing Skill#0 [w/GoldCrystalBanes] with Skill#1 [w/BondRingItems]
Line 793: Privateskill Jean replaced with Mauve skill (line 2 only)
BondRings started at line 841
line 845: Seliph ring's Battlewise+2 was replaced with a clone of Deirdre
line 859: Ring lilina was replaced with clone of Mia
line 869: Ring Edelgard replaced with a clone of Marica
line 871: Ring Claude replaced with a clone of ring Alfonse (line 1 only)
line 876: Ring Ryoma (line 2) replaced with clone of ring Anna (line 1 only)

These lines match with edits made in BondRingItem's Skill.xml
line 793: Mauve skill (line 2, modded)
line 845: Deirdre (modded)
line 859: Mia (Modded)
line 869: Marica (modded)
line 871: Alfonse (line 1, modded)
line 876: Anna (line 1, modded)

So I think the skills inserted by GoldCrystalBane shifted the indexes over, Cobalt correctly identified which lines were modded, but updated at the wrong indexes
GoldCrystalBane does it's skill insert after privateskill_anna (it was originally cloned from that one), who's one of the last characters. I made multiple privateskill edits in BondRingItems, but only mauves hit after the insert and then overlapped Jean
BondRingItems does it's skill insert after ringAnna, who is the last bondring.
The merged file then clones the last 5 bondrings: Marica, Heanius, Alfonse, Sharena, Anna (all lines lincluded)

Seems like this would work a lot better if I reversed merge order. Will try that next
